internal/api/v1beta1/role.go

```diff
@@ -81,6 +81,49 @@ func (h Handler) ListRoles(ctx context.Context, request *shieldv1beta1.ListRoles
 	return &shieldv1beta1.ListRolesResponse{Roles: roles}, nil
 }
 
+func (h Handler) CreateRole(ctx context.Context, request *shieldv1beta1.CreateRoleRequest) (*shieldv1beta1.CreateRoleResponse, error) {
+	logger := grpczap.Extract(ctx)
+	metaDataMap, err := metadata.Build(request.GetBody().GetMetadata().AsMap())
```

Comment:
What happens when `request.GetBody().GetMetadata()` is nil? Let's add a test for this?